torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -1685,58 +1685,58 @@ int dictLen(Stack& stack) {
   return 0;
 }
 
-int dictKeys(Stack& stack) {
-  auto dict = pop(stack).toGenericDict();
-  auto keys = c10::impl::GenericList(c10::impl::deprecatedUntypedList());
-  keys.reserve(dict.size());
-  for (auto& item : dict) {
-    keys.push_back(item.key());
-  }
-  push(stack, IValue(keys));
-  return 0;
-}
-
-template <typename Elem>
-c10::List<Elem> makeListForDictValues(
+template <unsigned int Index, typename Elem>
+c10::List<Elem> makeListForDictKeysOrValues(
     const std::vector<std::pair<IValue, IValue>>& order) {
   c10::List<Elem> values;
   values.reserve(order.size());
   for (const auto& item : order) {
-    values.push_back(item.second.to<Elem>());
+    values.push_back(std::get<Index>(item).template to<Elem>());
   }
   return values;
 }
 
-template <>
-c10::impl::GenericList makeListForDictValues<IValue>(
+template <unsigned int Index>
+c10::impl::GenericList makeGenericListForDictKeysOrValues(
     const std::vector<std::pair<IValue, IValue>>& order) {
   auto values = c10::impl::GenericList(c10::impl::deprecatedUntypedList());
```

Comment:
Is it possible to do `c10::List<IValue>` so we don't have to have a separate copy of this function for `GenericList`s?

Reply:
that doesn't compile. And you can't do partial template specialization so that you are only passing in the Index in